contracts/BC_fusion/GovToken.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20BurnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20PermitUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "./System.sol";
+
+interface IStakePool {
+    function getPooledBNB(address account) external view returns(uint256);
+}
+
+contract GovToken is
+    System,
+    Initializable,
+    ERC20Upgradeable,
+    ERC20BurnableUpgradeable,
+    ERC20PermitUpgradeable,
+    ERC20VotesUpgradeable
+{
+    string private constant NAME = "BSC Governance Token";
+    string private constant SYMBOL = "govBNB";
+
+    // validator StakePool contract => user => amount
+    mapping(address => mapping(address => uint256)) public mintedMap;
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize()
+    public
+    initializer
+    {
+        __ERC20_init(NAME, SYMBOL);
+        __ERC20Burnable_init();
+        __ERC20Permit_init(NAME);
+        __ERC20Votes_init();
+    }
+
+    function sync(address validatorPool, address account) external onlyStakeHub {
+        uint256 latestBNBAmount = IStakePool(validatorPool).getPooledBNB(account);
+        uint256 _mintedAmount = mintedMap[validatorPool][account];
+
+        if (_mintedAmount < latestBNBAmount) {
+            uint256 _needMint = latestBNBAmount - _mintedAmount;
+            mintedMap[validatorPool][account] += _needMint;
+            _mint(account, _needMint);
+        } else if (_mintedAmount > latestBNBAmount) {
+            uint256 _needBurn = _mintedAmount - latestBNBAmount;
+            mintedMap[validatorPool][account] -= _needBurn;
+            _burn(account, _needBurn);
+        }
+    }
+
+    function delegateVote(address delegator, address delegatee) external onlyStakeHub {
+        _delegate(delegator, delegatee);
+    }
+
+    function _transfer(address, address, uint256)
+    internal
+    pure
```

Comment:
why this is `pure`?

Reply:
because it is only `revert` , no storage query and changed